src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs

```diff
@@ -208,9 +208,22 @@ public byte[] Promote()
                 }
 
                 //Throw exception only if Transaction is still active and not yet aborted.
-                if (promoteException != null && Transaction.TransactionInformation.Status != TransactionStatus.Aborted)
+                if (promoteException != null)
                 {
-                    throw SQL.PromotionFailed(promoteException);
+                    try
```

Comment:
Also please log the `promoteException` in the event log as done here https://referencesource.microsoft.com/#System.Data/fx/src/data/System/Data/SqlClient/SqlDelegatedTransaction.cs,187
```
// example
SqlClientEventSource.Log.TryTraceEvent(promoteException)
```

Reply:
Sure, I think netcore in general lacks traces, there're present for .NET Framework:
https://github.com/dotnet/SqlClient/pull/1216/files/81ecb940d12b5ec9b9f1d388d8050801dd1a1c44#diff-2c3f43f080a5ac265aa69ad177fc583738bde1f2ff7b86d8bc83cab6ba7186c8L219

I'll try to bring parity in those missing areas.